src/component/App.tsx

```diff
@@ -1,58 +1,32 @@
-import { connect } from 'react-redux';
-import { Redirect, Route, Switch } from 'react-router-dom';
-import { RouteComponentProps } from 'react-router';
-
-import ProtectedRoute from './common/ProtectedRoute/ProtectedRoute';
-import LayoutPicker from './layout/LayoutPicker/LayoutPicker';
-
-import { routes } from './menu/routes';
-
-import styles from './styles.module.scss';
-
-import IAuthStatus from '../interfaces/user';
-import { useState, useEffect } from 'react';
-import NotFound from './common/NotFound/NotFound';
-import Feedback from './common/Feedback/Feedback';
-import SWRProvider from './providers/SWRProvider/SWRProvider';
 import ConditionallyRender from './common/ConditionallyRender';
 import EnvironmentSplash from './common/EnvironmentSplash/EnvironmentSplash';
+import Feedback from './common/Feedback/Feedback';
+import LayoutPicker from './layout/LayoutPicker/LayoutPicker';
 import Loader from './common/Loader/Loader';
-import useUser from '../hooks/api/getters/useUser/useUser';
+import NotFound from './common/NotFound/NotFound';
+import ProtectedRoute from './common/ProtectedRoute/ProtectedRoute';
+import SWRProvider from './providers/SWRProvider/SWRProvider';
 import ToastRenderer from './common/ToastRenderer/ToastRenderer';
+import styles from './styles.module.scss';
+import { Redirect, Route, Switch } from 'react-router-dom';
+import { RouteComponentProps } from 'react-router';
+import { routes } from './menu/routes';
+import { useAuth } from '../hooks/api/getters/useAuth/useAuth';
+import { useEffect } from 'react';
 
 interface IAppProps extends RouteComponentProps {
-    user: IAuthStatus;
-    fetchUiBootstrap: any;
-    feedback: any;
+    fetchUiBootstrap: () => void;
 }
-const App = ({ location, user, fetchUiBootstrap }: IAppProps) => {
-    // because we need the userId when the component load.
-    const { splash, user: userFromUseUser, authDetails } = useUser();
 
-    const [showSplash, setShowSplash] = useState(false);
-    const [showLoader, setShowLoader] = useState(false);
-    useEffect(() => {
-        fetchUiBootstrap();
-        /* eslint-disable-next-line */
-    }, [user.authDetails?.type]);
+export const App = ({ fetchUiBootstrap }: IAppProps) => {
+    const { auth, refetchAuth } = useAuth();
```

Comment:
I'm not sure how we should name this. This name suggests that it contains authentication details, but it's actually encompassing different data at different points in time: 

(1) If the user is not authenticated, the call will return authDetails which is a set of data used to render the different login options in the login page
(2) If the user is authentication, then the API endpoint returns user _metadata_ and user _authorization_ in the form of permission. 

I think this might be a flaw in our API design, but perhaps we could alleviate this a bit by creating abstractions on top of the base useSWR hook. 

I'm thinking something along the lines of: 

```
useUserEndpoint -> Base hook
useAuthentication -> Use the useUserEndpoint and expose authDetails
useUser -> Use the useUserEndpoint and expose user details
useAuthorization -> Use the useUserEndpoint and expose user permissions
```

Since all hooks will use the same SWR cache there should be no degradation of performance doing it this way, and it might be easier to understand in different contexts of the code.  Thoughts?

Reply:
Good idea! Pushed a change which splits the hooks. 

Still not sure about the naming, but I like that they share a prefix since they fetch from the same endpoint.